Sources/Firebase/API/Routes/DynamicLinksRoutes.swift

```diff
@@ -0,0 +1,24 @@
+import Vapor
+
+public protocol DynamicLinksRoutes {
+    func createShortLink(param: DynamicLinkRequest) throws -> Future<DynamicLink>
+}
+
+public struct FirebaseDynamicLinksRoutes: DynamicLinksRoutes {
+
+    private let request: FirebaseRequest
+    
+    init(request: FirebaseRequest) {
+        self.request = request
+    }
+
+    /// Create short links
+    /// [Learn More →](https://firebase.google.com/docs/reference/dynamic-links/link-shortener)
+    public func createShortLink(param: DynamicLinkRequest) throws -> Future<DynamicLink> {
+        return try request.send(method: .POST,
+                                url: FirebaseAPIEndpoint.DynamicLinks.shortLinks.endpoint,
+                                query: "",
+                                body: HTTPBody(string: param.toEncodedBody()),
```

Comment:
I failed. I added an extension to add `LosslessHTTPBodyRepresentable` to `DynamicLinkRequest`. Then I implement the `convertToHTTPBody()` method:

```
extension DynamicLinkRequest: LosslessHTTPBodyRepresentable {
    public func convertToHTTPBody() -> HTTPBody {
        return HTTPBody(string: self.toEncodedBody())
    }
}
```

but the `self.toEncodedBody()` throws an exception and I can't propagate this exception because `convertToHTTPBody` doesn't throws exception. 

This implementation seems more complex than current one...

Reply:
ok